### PR TITLE
fix(api,app,shared-data): Fix LPC for protocols using the plate reader module.

### PR DESCRIPTION
**Pull request by vegano1**
# Overview

This fixes an issue when attempting to access the module state store to get the plate reader state when the `load_module` protocol context command is used, since the module state is not created at this point it would lead to a `ModuleNotLoadedError` error. Also, you cannot LPC any labware if a Plate Reader is loaded in the protocol since the labware lid is a labware type. You also cannot LPC any labware loaded on the Plate Reader itself, so for now let's disable LPC for the plate reader to unlock LPC everywhere else.

Do not merge before merging in the TC lids pr [Opentrons/opentrons#16345](https://github.com/Opentrons/opentrons/pull/16345) which adds `allowedRoles` support.

Closes: [PLAT-542](https://opentrons.atlassian.net/browse/PLAT-542)

## Test Plan and Hands on Testing

- [x] Make sure you can perform LPC on any other labware when a Plate Reader is loaded in a protocol

```
from typing import cast
from opentrons import protocol_api
from opentrons.protocol_api.module_contexts import AbsorbanceReaderContext

# metadata
metadata = {
    'protocolName': 'Absorbance Reader Multi read test',
    'author': 'Platform Expansion',
}

requirements = {
    "robotType": "Flex",
    "apiLevel": "2.21",
}

# protocol run function
def run(protocol: protocol_api.ProtocolContext):
    mod = cast(AbsorbanceReaderContext, protocol.load_module("absorbanceReaderV1", "C3"))
    plate = protocol.load_labware("nest_96_wellplate_200ul_flat", "C2")
    tiprack_1000 = protocol.load_labware(load_name='opentrons_flex_96_tiprack_1000ul', location="B2")
    trash_labware = protocol.load_trash_bin("B3")
    instrument = protocol.load_instrument("flex_8channel_1000", "right", tip_racks=[tiprack_1000])
    instrument.trash_container = trash_labware

    # pick up tip and perform action
    instrument.pick_up_tip(tiprack_1000.wells_by_name()['A1'])
    instrument.aspirate(100, plate.wells_by_name()['A1'])
    instrument.dispense(100, plate.wells_by_name()['B1'])
    instrument.return_tip()
   
    # Initialize to a single wavelength with reference wavelength
    mod.initialize('single', [600], 450)

    # NOTE: CANNOT INITIALIZE WITH THE LID OPEN

    # Remove the Plate Reader lid using the Gripper.
    mod.open_lid()
    protocol.move_labware(plate, mod, use_gripper=True)
    mod.close_lid()

    # Take a reading and show the resulting absorbance values.
    result = mod.read()
    msg = f"single: {result}"
    protocol.comment(msg=msg)
    protocol.pause(msg=msg)

    # Initialize to multiple wavelengths
    protocol.pause(msg="Perform Multi Read")
    mod.open_lid()
    protocol.move_labware(plate, "C2", use_gripper=True)
    mod.close_lid()
    mod.initialize('multi', [450, 570, 600])

    # Open the lid and move the labware into the reader
    mod.open_lid()
    protocol.move_labware(plate, mod, use_gripper=True)

    # pick up tip and perform action on labware inside plate reader
    instrument.pick_up_tip(tiprack_1000.wells_by_name()['A1'])
    instrument.aspirate(100, plate.wells_by_name()['A1'])
    instrument.dispense(100, plate.wells_by_name()['B1'])
    instrument.return_tip()

    mod.close_lid()

    # Take reading
    result = mod.read()
    msg = f"multi: {result}"
    protocol.comment(msg=msg)
    protocol.pause(msg=msg)

    # Place the Plate Reader lid back on using the Gripper.
    mod.open_lid()
    protocol.move_labware(plate, "C2", use_gripper=True)
    mod.close_lid()

```

## Changelog

- Catch `ModuleNotLoadedError` in the `load_module` command for loading the plate reader lid during LPC
- Omit the plate reader lid and any labware on the plate reader from requiring LPC in the client.
- Add `allowedRoles` key and `lid` type to the plate reader definition so the front-end can filter out this type of geometry, 

## Review requests

- Does this make sense?
- Any other ways I should be going for omitting labware on the client?

## Risk assessment
Low, plate reader work

[PLAT-542]: https://opentrons.atlassian.net/browse/PLAT-542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shared-data/labware/definitions/2/opentrons_flex_lid_absorbance_plate_reader_module/1.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
   "namespace": "opentrons",
   "version": 1,
   "schemaVersion": 2,
-  "allowedRoles": ["fixture"],
+  "allowedRoles": ["fixture", "lid"],
```
Comment (smb2268) [inline shared-data/labware/definitions/2/opentrons_flex_lid_absorbance_plate_reader_module/1.json:35]: yay 🧢

**Review by smb2268 — APPROVED** [review]
Changes lgtm! I will implement the final LPC solution in this ticket https://opentrons.atlassian.net/browse/PLAT-483

File: app/src/organisms/LabwarePositionCheck/useLaunchLPC.tsx [hunk 2]
```diff
@@ -61,11 +61,14 @@ export function useLaunchLPC(
...
           getLabwareDefinitionsFromCommands(
             mostRecentAnalysis?.commands ?? []
-          ).map(def =>
-            createLabwareDefinition({
-              maintenanceRunId: maintenanceRun?.data?.id,
-              labwareDef: def,
-            })
+          ).map(def => {
+            // HACK: Skip plate reader for now
+            if (def.parameters.loadName !== 'opentrons_flex_lid_absorbance_plate_reader_module')
```
Comment (CaseyBatten) [inline app/src/organisms/LabwarePositionCheck/useLaunchLPC.tsx]: Based on changes in my [PR](https://github.com/Opentrons/opentrons/pull/16345) you'll want to ensure that `'opentrons_flex_lid_absorbance_plate_reader_module'` is moved from here to the list of `filteredLabware` alongside the TC lids.

File: app/src/organisms/LabwarePositionCheck/utils/getProbeBasedLPCSteps.ts [hunk 3]
```diff
@@ -73,7 +73,12 @@ function getAllCheckSectionSteps(
     []
   )
 
-  return labwareLocations.map(
+  // HACK: Remove LPC for plate reader to unblock science.
```
Comment (CaseyBatten) [inline app/src/organisms/LabwarePositionCheck/utils/getProbeBasedLPCSteps.ts:76]: Should be safe for our current needs, and given the limited use cases and labware types we will be allowing on the plate reader slot anyways I don't think theres an issue in preventing LPC on said labware.

**Review by CaseyBatten — APPROVED** [review]
This looks good, and lines up with the solution on the TC lid side of things for the purposes of unblocking our desired behavior for now.

**Pull request merged**